core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/EditorAgent.java

```diff
@@ -63,10 +62,19 @@
     /**
      * Get all opened editors
      *
-     * @return map with all opened editors
+     * @return list with all opened editors
      */
     @NotNull
-    Map<String, EditorPartPresenter> getOpenedEditors();
+    List<EditorPartPresenter> getOpenedEditors();
+
+    /**
+     * Get opened editor by related file path
+     *
+     * @param path path of the file opened in editor
+     * @return opened editor or null if it does not exist
+     */
+    @Nullable
+    EditorPartPresenter getOpenedEditor(String path);
```

Comment:
Then use [Path.java](https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/resource/Path.java). We have specialized class to work with paths.